server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAOAWSCryptoTest.java

```diff
@@ -84,4 +85,58 @@ public BlobStore testee() {
     public BlobId.Factory blobIdFactory() {
         return BLOB_ID_FACTORY;
     }
+
+    @Override
+    @Disabled("JAMES-2829 Not supported yet")
+    public void deleteShouldNotThrowWhenBlobDoesNotExist() {
```

Comment:
to avoid all these @Disabled, you can simply add new tests into sub-contracts.
That way, you add the contract only to supporting implementation

Reply:
True it's ok temporarily, but when we will have all the implementations it will make little sense I think. But I will refactor then

Reply:
> True it's ok temporarily, but when we will have all the implementations it will make little sense I think.

Not sure about that: grouping things is rarely a waste of time

Reply:
Yes but I meant, at the end `delete` is an operation of the blobStore like `read`, `save`, etc... Thus I think it's better to have them grouped in the same contract, and not having a contract for each method? So I think when we get all the implementations working with delete we can refactor the contracts